src/linux_statx.hpp

```diff
@@ -0,0 +1,171 @@
+//  linux_statx.hpp  --------------------------------------------------------------------//
+
+//  Copyright 2002-2009, 2014 Beman Dawes
+//  Copyright 2019 Andrey Semashev
+//  Copyright 2020 Seyyed Soroosh Hosseinalipour
+
+//  Distributed under the Boost Software License, Version 1.0.
+//  See http://www.boost.org/LICENSE_1_0.txt
+
+//  See library home page at http://www.boost.org/libs/filesystem
+
+//--------------------------------------------------------------------------------------//
+
+#ifndef BOOST_FILESYSTEM3_SRC_LINUX_STATX_HPP_
+#define BOOST_FILESYSTEM3_SRC_LINUX_STATX_HPP_
+
+// Only for internal use
+// Copy many contents from stat.h
```

Comment:
I don't like this. Please, don't reimplement libc. Aside from maintenance cost, it is also unclear from the licensing perspective (you cannot copy/paste third party code and relicense it under BSL). If libc does not implement the necessary API then make Boost.Filesystem return `ENOSYS`.

Reply:
@Lastique  I think it is very useful to implement internal implementation of statx syscall when glibc does not supported and it is cause to our library is more useful, but as you wish I remove it.